projects/demo-integrations/cypress/tests/core/data-list/data-list.spec.ts

```diff
@@ -1,18 +1,69 @@
-import {DEFAULT_TIMEOUT_BEFORE_ACTION} from '../../../support/shared.entities';
+import {
+    DEFAULT_TIMEOUT_BEFORE_ACTION,
+    EXAMPLE_ID,
+} from '../../../support/shared.entities';
 
 describe('DataList', () => {
     beforeEach(() => {
         cy.tuiVisit('/components/data-list');
     });
 
+    it('Custom list', () => {
+        cy.get('#custom tui-select').should('be.visible').click();
+        cy.get('tui-dropdown-box').should('be.visible').matchImageSnapshot('1-#custom');
+    });
+
+    it('Links', () => {
+        cy.get(`#links [automation-id=${EXAMPLE_ID}] tui-hosted-dropdown`).click();
+        cy.get('tui-dropdown-box').should('be.visible').matchImageSnapshot('2-#links');
+    });
+
+    it('Submenu', () => {
+        // 300ms - debounce time of DataListDropdownManager + 100ms for flaky-free test
+        const DATA_LIST_DEBOUNCE_WAIT = 400;
+
+        cy.get(`#submenu [automation-id=${EXAMPLE_ID}] tui-hosted-dropdown`)
+            .scrollIntoView()
+            .should('be.visible')
+            .click();
+        cy.get('body')
+            .type('{downarrow}'.repeat(2))
+            .wait(DATA_LIST_DEBOUNCE_WAIT)
+            .type('{rightarrow}')
+            .type('{downarrow}'.repeat(3))
+            .wait(DATA_LIST_DEBOUNCE_WAIT)
+            .type('{rightarrow}')
+            .type('{downarrow}'.repeat(2));
+
+        cy.matchImageSnapshot('3-#submenu', {
+            capture: 'viewport',
+            blackout: ['tui-doc-navigation'],
+        });
+    });
+
+    it('Form control', () => {
+        cy.get(`#control [automation-id=${EXAMPLE_ID}] tui-hosted-dropdown`)
+            .scrollIntoView()
+            .should('be.visible')
+            .click();
+
+        cy.get('tui-dropdown-box').should('be.visible').matchImageSnapshot('4-#control');
+    });
+
     it('Complex', {responseTimeout: 30_000}, () => {
         const demo = `tui-doc-example[id=complex] .t-demo`;
 
         cy.get(demo).scrollIntoView().should('be.visible');
-        cy.get('tui-doc-page').matchImageSnapshot('1-data-list-initial');
+        cy.matchImageSnapshot('5-1-data-list-initial', {
```

Comment:
it is LESS expensive than taking the screenshot of the whole `tui-doc-page`

Reply:
```ts
cy.get('tui-doc-page').matchImageSnapshot('1-data-list-initial');
```
<img width="300" src="https://user-images.githubusercontent.com/35179038/170482951-7097dd1b-bd53-4a4e-bab5-7d82eb8d1a98.png">

Reply:
new version
![5-2-data-list-opened snap](https://user-images.githubusercontent.com/35179038/170483148-9a1dd60d-9e7a-446c-85b2-d32f253f0d46.png)